vicodyn/src/gateway/vicodyn.cpp

```diff
@@ -133,25 +116,47 @@ vicodyn_t::vicodyn_t(context_t& _context, const std::string& _local_uuid, const
         dynamic_t result = dynamic_t::empty_object;
         dynamic_t& app_result = result.as_object()["apps"];
         peers.apply_shared([&](const vicodyn::peers_t::data_t& data) mutable {
+            auto convert_app_services = [&](const vicodyn::peers_t::app_services_t& from) {
+                dynamic_t app_services_result = dynamic_t::empty_object;
+                for (const auto& app_service : from) {
+                    dynamic_t& peer_result = app_services_result.as_object()[app_service.first];
+
+                    auto banned_still_ns = app_service.second.banned_still().count();
+                    if (banned_still_ns > 0) {
+                        peer_result.as_object()["banned_ms"] = banned_still_ns / 1000000;
+                    }
+
+                    auto request_duration_ns = std::max(1., app_service.second.avg_request_duration_ns());
+                    peer_result.as_object()["request_duration_ms"] = request_duration_ns / 1000000;
+
+                    auto system_weight = vicodyn::defaults::system_weight;
+                    auto peer_it = data.peers.find(app_service.first);
+                    if (peer_it != data.peers.end()) {
+                        system_weight = peer_it->second.system_weight.load();
+                    }
+                    peer_result.as_object()["system_weight"] = system_weight;
+                    peer_result.as_object()["total_weight"] = system_weight / request_duration_ns;
+                    peer_result.as_object()["x_cocaine_cluster"] = peer_it->second.peer->x_cocaine_cluster();
+                }
+                return app_services_result;
+            };
+
             if(app.empty()) {
-                app_result = data.apps;
+                for (const auto& app : data.apps) {
+                    app_result.as_object()[app.first] = convert_app_services(app.second);
+                }
             } else {
                 auto it = data.apps.find(app);
                 if(it != data.apps.end()) {
                     app_result = dynamic_t::empty_object;
-                    app_result.as_object()[it->first] = it->second;
+                    app_result.as_object()[it->first] = convert_app_services(it->second);
                 }
             }
         });
         return result;
     });
-    d->on<io::vicodyn::info>([&]() {
-        dynamic_t result = dynamic_t::empty_object;
-        peers.apply_shared([&](const vicodyn::peers_t::data_t& data) {
-            result.as_object()["apps"] = data.apps;
-            result.as_object()["peers"] = data.peers;
-        });
-        return result;
+    d->on<io::vicodyn::info>([]() {
+        return dynamic_t("not implemented");
```

Comment:
May be throw exception to client? But don't remember whether it possible from simple lambda based handler. Forget if not possible.

Reply:
Fixed